Automattic/wp-calypso
# Site Assembler: Render patterns on demand

Related to https://github.com/Automattic/wp-calypso/issues/72548

## Proposed Changes

* Render the specified patterns when the pattern selector is open
* Render the list of patterns by `useAsyncList` to render the patterns one by one. The `useAsyncList` hook has an option called `step` to control how many items you want to render every time. The default value is 1 and I think it's enough for now.
* Preload the styles assets as much as possible so that the styles will be loaded when the user enters the site assembler for the first time

https://user-images.githubusercontent.com/13596067/217470562-9f1ce6ca-c67a-48fe-973d-914d1798abf8.mov

## Testing Instructions



* Go to `/setup?siteSlug=<your_site>`
* Continue until you land on the Design Picker
* On the Design Picker screen, scroll down to the bottom and select BCPA CTA
* On the Pattern Assembler screen, select any patterns to see whether it's faster than before

## Pre-merge Checklist
